src/main/java/com/yelp/nrtsearch/server/luceneserver/state/BackendGlobalState.java

```diff
@@ -175,21 +226,25 @@ public synchronized void deleteIndex(String name) throws IOException {
         immutableState.persistentGlobalState.asBuilder().withIndices(updatedIndexInfoMap).build();
     stateBackend.commitGlobalState(updatedState);
 
-    immutableState = new ImmutableState(updatedState, new HashMap<>(immutableState.indexStateMap));
+    IndexStateManager stateManager = immutableState.indexStateManagerMap.get(name);
+    Map<String, IndexStateManager> updatedIndexStateManagerMap =
+        new HashMap<>(immutableState.indexStateManagerMap);
+    updatedIndexStateManagerMap.remove(name);
+
+    immutableState = new ImmutableState(updatedState, updatedIndexStateManagerMap);
+    stateManager.close();
   }
 
   @Override
   public synchronized void indexClosed(String name) {
-    Map<String, IndexState> updatedIndexStateMap = new HashMap<>(immutableState.indexStateMap);
-    updatedIndexStateMap.remove(name);
-    immutableState = new ImmutableState(immutableState.persistentGlobalState, updatedIndexStateMap);
+    throw new UnsupportedOperationException();
```

Comment:
This method is only called by the `LegacyIndexState` as part of the `close` method. Since this global state implementation is no longer used with the legacy state, the functionality is no longer needed.